app/src/main/java/org/simple/clinic/widgets/ThreeTenBpDatePickerDialog.kt

```diff
@@ -0,0 +1,43 @@
+package org.simple.clinic.widgets
+
+import android.app.DatePickerDialog
+import android.app.DatePickerDialog.OnDateSetListener
+import android.content.Context
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.toUtcInstant
+import org.threeten.bp.LocalDate
+
+/**
+ * An extension of [DatePickerDialog] that supports the ThreeTenBp
+ * classes.
+ *
+ * The DatePickerDialog uses 0-based indices for the Month (0 is January, 1 is February...),
+ * while LocalDate uses 1-based indices (1 is January, 2 is February...).
+ *
+ * So when we convert from LocalDate to the DatePicker (and vice versa), we have to adjust the
+ * month accordingly.
+ **/
+class ThreeTenBpDatePickerDialog(
+    context: Context,
+    preselectedDate: LocalDate,
+    allowedDateRange: ClosedRange<LocalDate>,
+    clock: UserClock,
+    datePickedListener: (LocalDate) -> Unit
+) : DatePickerDialog(
```

Comment:
Now that we have added MaterialComponents, should we try and use the material date picker, WDYT?

https://material.io/components/pickers/#

Reply:
I tried doing this, but the change is too invasive to use at the moment. I'll take a look at this later once I complete the feature.